internal/kibana/slo_test.go

```diff
@@ -0,0 +1,266 @@
+				SkipFunc: versionutils.CheckIfVersionIsUnsupported(minSupportedVersion),
+				Config:   getTFConfig(sloName, "sli.apm.transactionDuration", true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "settings.0.sync_delay", "5m"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "settings.0.frequency", "5m"),
+				),
+			},
+			{
+				SkipFunc: versionutils.CheckIfVersionIsUnsupported(minSupportedVersion),
+				Config:   getTFConfig(sloName, "sli.apm.transactionErrorRate", true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.type", "sli.apm.transactionErrorRate"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.environment", "production"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.service", "my-service"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.transaction_type", "request"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.transaction_name", "GET /sup/dawg"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.index", "my-index"),
+				),
+			},
+			{
+				SkipFunc: versionutils.CheckIfVersionIsUnsupported(minSupportedVersion),
+				Config:   getTFConfig(sloName, "sli.kql.custom", true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.type", "sli.kql.custom"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.index", "my-index"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.good", "latency < 300"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.total", "*"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.filter", "labels.groupId: group-0"),
+					resource.TestCheckResourceAttr("elasticstack_kibana_slo.test_slo", "indicator.0.params.0.timestamp_field", "custom_timestamp"),
+				),
+			},
+			{
+				SkipFunc:    versionutils.CheckIfVersionIsUnsupported(minSupportedVersion),
+				Config:      getTFConfig(sloName, "sli.histogram.custom", true),
+				ExpectError: regexp.MustCompile("the sli.histogram.custom indicator type is currently unsupported"),
+			},
+			{
+				SkipFunc:    versionutils.CheckIfVersionIsUnsupported(minSupportedVersion),
+				Config:      getTFConfig(sloName, "sli.metric.custom", true),
+				ExpectError: regexp.MustCompile("the sli.metric.custom indicator type is currently unsupported"),
+			},
+		},
+	})
+}
+
+func getTFConfig(name string, indicatorType string, settingsEnabled bool) string {
```

Comment:
The name is too short for such a specific function that is used only in one test. IMO it should be renamed to be more verbose or, ideally, moded to the test function scope.

Reply:
Yeah good point. Happy to move it in the test scope.